src/diffusers/models/unets/unet_motion_model.py

```diff
@@ -450,6 +454,36 @@ def from_unet2d(
         model.time_proj.load_state_dict(unet.time_proj.state_dict())
         model.time_embedding.load_state_dict(unet.time_embedding.state_dict())
 
+        if any(
+            isinstance(proc, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0))
+            for proc in unet.attn_processors.values()
+        ):
+            attn_procs = {}
+            for name, processor in unet.attn_processors.items():
+                if name.endswith("attn1.processor"):
+                    attn_processor_class = (
+                        AttnProcessor2_0 if hasattr(F, "scaled_dot_product_attention") else AttnProcessor
+                    )
+                    attn_procs[name] = attn_processor_class()
+                else:
+                    attn_processor_class = (
+                        IPAdapterAttnProcessor2_0
+                        if hasattr(F, "scaled_dot_product_attention")
+                        else IPAdapterAttnProcessor
+                    )
+                    attn_procs[name] = attn_processor_class(
+                        hidden_size=processor.hidden_size,
+                        cross_attention_dim=processor.cross_attention_dim,
+                        scale=processor.scale,
+                        num_tokens=processor.num_tokens,
+                    )
+            for name, processor in model.attn_processors.items():
+                if name not in attn_procs:
+                    attn_procs[name] = processor.__class__()
+            model.set_attn_processor(attn_procs)
```

Comment:
Is it the same as this?
```python
model.set_attn_processor(unet.attn_processors)
```

Reply:
actually no, because the UNetMotion used in AnimateDiff has motion modules, that the original pipeline does not have

if you do something like this:

```py
if any(
    isinstance(proc, (IPAdapterAttnProcessor, IPAdapterAttnProcessor2_0))
    for proc in unet.attn_processors.values()
    ):
    model.set_attn_processor(unet.attn_processors)
    model.config.encoder_hid_dim_type = "ip_image_proj"
    model.encoder_hid_proj = unet.encoder_hid_proj
```
you will end up (in the particular case of the code snippet above) with a ValueError, because the number of attention processors does not match
`
ValueError: A dict of processors was passed, but the number of processors 32 does not match the number of attention layers: 74. Please make sure to pass 74 processor classes.
`

Reply:
@yiyixuxu I don't know if you read it, I'm pinging you because the issue has gone stale
thank you

Reply:
hey, yes, indeed, I missed this one!
thanks for pining!